openmc/model/model.py

```diff
@@ -1439,3 +1442,492 @@ def differentiate_mats(self, diff_volume_method: str = None, depletable_only: bo
             self.materials = openmc.Materials(
                 self.geometry.get_all_materials().values()
             )
+
+    def _generate_infinite_medium_mgxs(self, groups, nparticles, mgxs_fname) -> None:
+        """Generate a MGXS library by running multiple OpenMC simulations, each
+        representing an infinite medium simulation of a single isolated
+        material. A discrete source is used to sample particles, with an equal
+        strength spread across each of the energy groups. This is a highly naive
+        method that ignores all spatial self shielding effects and all resonance
+        shielding effects between materials.
+
+        Parameters
+        ----------
+        groups : openmc.mgxs.EnergyGroups
+            Energy group structure for the MGXS.
+        mgxs_fname : str
+            Filename for the MGXS HDF5 file.
+        """
+        warnings.warn(
+            "The infinite medium method of generating MGXS may hang if a material has a k-infinity > 1.0.")
+        mgxs_sets = []
+        for material in self.materials:
+            openmc.reset_auto_ids()
+            model = openmc.Model()
+
+            # Name
+            name = material.name
+            print(f"Running infinite medium calculation for material: {name}")
+            # Material
+            model.materials = [material]
+
+            # Settings
+            model.settings.batches = 100
+            model.settings.particles = nparticles
+            model.settings.run_mode = 'fixed source'
+
+            # Make a discrete source that is uniform over the bins of the group structure
+            n_groups = groups.num_groups
+            midpoints = []
+            strengths = []
+            for i in range(n_groups):
+                bounds = groups.get_group_bounds(i+1)
+                midpoints.append((bounds[0] + bounds[1]) / 2.0)
+                strengths.append(1.0)
+
+            energy_distribution = openmc.stats.Discrete(
+                x=midpoints, p=strengths)
+            model.settings.source = [openmc.IndependentSource(
+                space=openmc.stats.Point(), energy=energy_distribution, strength=1.0)]
+            model.settings.output = {'summary': True, 'tallies': False}
+
+            # Geometry
+            box = openmc.model.RectangularPrism(
+                100000.0, 100000.0, boundary_type='reflective')
```

Comment:
I'm not sure if these "magic numbers" will become an issue, but is there a `model.max_extents` parameter that could be used for this instead? I can't imagine anyone wanting a larger simulation space, but it moves the magic into a parameter, at least.

Reply:
Interesting -- I appreciate the rundown!